lib/Extension/LanguageServerPhpCsFixer/LspCommand/FormatCommand.php

```diff
@@ -35,7 +35,14 @@ public function __invoke(string $uri, ?array $rules = null): Promise
             $textDocument = $this->workspace->get($uri);
 
             $rulesOpt = $rules ? ['--rules', ...$rules] : [];
-            $diff = yield $this->phpCsFixer->fix($textDocument->text, ['--diff', '--dry-run', ...$rulesOpt]);
+
+            $configOpt = [];
+            if ($this->phpCsFixer->hasConfigPath()) {
+                $rulesOpt = [];
+                $configOpt = ['--config', $this->phpCsFixer->getConfigPath()];
```

Comment:
this should be in the `PhpCsFixerProcess` class. The "config path" should not escape from the `PhpCsFixerProcess`

Reply:
we can't or we need to add option ignore config as in diagnostics there are processed per rule. Config will overwrite. What do you think?

Reply:
confused, shouldn't the config be the same regardless of where it's invoked?

Reply:
the flow of diagnostics is like that in short:
- get all errors/warnings in json
- get rules from json
- go through all rules and format them (so, no config with all rules)

Maybe I am wrong or the flow is wrong but for me it works. Maybe we can get descriptions in other way, then we could ignore this run per rule and give error/warning in different way

Reply:
I still don't see why we'd want to have "no config" in one situation and not in another. It's using the `.php_cs_.dist.php` file (or whatever) by default anyway - so this option is changing the location it's looking for that file basically?

Reply:
default file is `.php-cs-fixer.dist.php` in root directory, but good practice is to use `.php-cs-fixer.php` (added to `.gitignore`). So, yes, this `--config` is to locate configuration. 

Before I said that there is overwrittings. I was wrong:
 
```
[PhpCsFixer\ConfigurationException\InvalidConfigurationException (16)]
  Passing both `--config` and `--rules` options is not allowed.
```

Reply:
There is `FormatCommand` what takes `$rules` as argument. I don't get when it is used. If it doesn't take argument then everything can be simplified: on `fix()` check does options contains rules and if not - try to add config

Reply:
OK. So, tried to move into `PhpCsFixerProcess`. For my workflow: show diagnostics on the fly and fix on demand - it works. Config path is used and it listens to its rules.